recon_test_pack/run_test_listmode_recon.sh

```diff
@@ -0,0 +1,141 @@
+#! /bin/sh
+# A script to check to see if reconstruction of simulated data gives the expected result.
+#
+#  Copyright (C) 2011 - 2011-01-14, Hammersmith Imanet Ltd
+#  Copyright (C) 2011-07-01 - 2011, Kris Thielemans
+#  Copyright (C) 2014, University College London
+#  This file is part of STIR.
+#
+#  This file is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+#  This file is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  See STIR/LICENSE.txt for details
+#      
+# Author Nikos Efthimiou
+#
+
+# Scripts should exit with error code when a test fails:
+if [ -n "$TRAVIS" ]; then
+    # The code runs inside Travis
+    set -e
+fi
+
+echo This script should work with STIR version ">"3.0. If you have
+echo a later version, you might have to update your test pack.
+echo Please check the web site.
+echo
+
+#
+# Options
+#
+MPIRUN=""
+
+#
+# Parse option arguments (--)
+# Note that the -- is required to suppress interpretation of $1 as options 
+# to expr
+#
+while test `expr -- "$1" : "--.*"` -gt 0
+do
+
+  if test "$1" = "--mpicmd"
+  then
+    MPIRUN="$2"
+    shift 1
+  elif test "$1" = "--help"
+  then
+    echo "Usage: `basename $0` [--mpicmd somecmd] [install_dir]"
+    echo "(where [] means that an argument is optional)"
+    echo "See README.txt for more info."
+    exit 1
+  else
+    echo Warning: Unknown option "$1"
+    echo rerun with --help for more info.
+    exit 1
+  fi
+
+  shift 1
+
+done 
+
+if [ $# -eq 1 ]; then
+  echo "Prepending $1 to your PATH for the duration of this script."
+  PATH=$1:$PATH
+fi
+
+# first delete any files remaining from a previous run
+rm -f my_*v my_*s my_*S
+
+echo "=== calculate background data"
+echo "===== randoms"
+
+echo "====== create delayed fansums"
+INPUT=PET_ACQ_small.l.hdr.STIR OUTPUT=my_fansums_delayed ${INSTALL_DIR}lm_fansums lm_fansums_delayed.par 2>my_fansums.log
+
+echo "====== estimate singles from fansums"
+niters=10
+# Note: the last 2 numbers are specific to the mMR
+${INSTALL_DIR}find_ML_singles_from_delayed -f my_MLsingles_f1 my_fansums_delayed_f1.dat  $niters 2 343 </dev/null
+
+echo " ===== estimate randoms from singles"
+${INSTALL_DIR}construct_randoms_from_singles my_MLrandoms_f1 my_MLsingles_f1 Siemens_mMR_seg2.hs $niters 2>my_construct_randoms_from_singles.log
+
+echo "=== simulate normalisation data"
+# For normalisation data we are going to use a cylinder in the center,
+# with water attenuation values
+
+echo "===  make fake emission image"
```

Comment:
cleaner if we'd immediately generate the atten image. This stuff confuses people (I know you copied it from mine. sorry!)

Reply:
👍